### PR TITLE
Hw controllers add

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

**Review by KovalAlexey — APPROVED** [review]

File: src/main/java/com/cinema/shop/controllers/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private OrderService orderService;
+    private UserService userService;
+    private ShoppingCartService shoppingCartService;
+    private OrderMapper orderMapper;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/controllers/OrderController.java]: add `final`

File: src/main/java/com/cinema/shop/controllers/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private ShoppingCartService shoppingCartService;
+    private MovieSessionService movieSessionService;
+    private UserService userService;
+    private ShoppingCartMapper shoppingCartMapper;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/controllers/ShoppingCartController.java]: same

File: src/main/java/com/cinema/shop/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private UserService userService;
+    private UserMapper userMapper;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/controllers/UserController.java]: same

File: src/main/java/com/cinema/shop/dao/TicketDao.java [hunk 4]
```diff
@@ -4,4 +4,6 @@
 
 public interface TicketDao {
     Ticket add(Ticket ticket);
+
+    Ticket getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/dao/TicketDao.java]: do you use it?

File: src/main/java/com/cinema/shop/dao/UserDao.java [hunk 5]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    User getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/dao/UserDao.java:11]:
```suggestion
    User get(Long id);
```
remember about uniform method names

File: src/main/java/com/cinema/shop/model/dto/ShoppingCartResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.shop.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/model/dto/ShoppingCartResponseDto.java]: aren't `cartId` and `userId` the same?

File: src/main/java/com/cinema/shop/model/dto/UserResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.shop.model.dto;
+
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
+    private Long id;
+    private String email;
+    private String password;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/model/dto/UserResponseDto.java]: let's not return secure data

File: src/main/java/com/cinema/shop/service/TicketService.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.cinema.shop.service;
+
+import com.cinema.shop.model.Ticket;
+
+public interface TicketService {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/service/TicketService.java]: do you need this service?

File: src/main/java/com/cinema/shop/service/UserService.java [hunk 9]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    User getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/service/UserService.java:11]:
```suggestion
    User get(Long id);
```
again, uniform method names everywhere

File: src/main/java/com/cinema/shop/mappers/UserMapper.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.cinema.shop.model.User;
+import com.cinema.shop.model.dto.UserRequestDto;
+import com.cinema.shop.model.dto.UserResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
+
+    public User getUserFromDto(UserRequestDto userRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/mappers/UserMapper.java:11]: where do you use this method?

File: src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.cinema.shop.model.ShoppingCart;
+import com.cinema.shop.model.dto.ShoppingCartResponseDto;
+import com.cinema.shop.service.UserService;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartMapper {
+    private UserService userService;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java]: do you use this dependency?

File: src/main/java/com/cinema/shop/mappers/OrderMapper.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+import com.cinema.shop.model.Order;
+import com.cinema.shop.model.dto.OrderResponseDto;
+import com.cinema.shop.service.UserService;
+import java.time.format.DateTimeFormatter;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OrderMapper {
+    private UserService userService;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/mappers/OrderMapper.java]: same

File: src/main/java/com/cinema/shop/model/dto/OrderResponseDto.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
+package com.cinema.shop.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketIds;
+    private String orderData;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/model/dto/OrderResponseDto.java]:
```suggestion
    private String orderDate;
```

File: src/main/java/com/cinema/shop/mappers/OrderMapper.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+    public OrderMapper(UserService userService) {
+        this.userService = userService;
+    }
+
+    public OrderResponseDto getOrderDto(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        orderResponseDto.setId(order.getId());
+        orderResponseDto.setTicketIds(order.getTickets().stream()
+                .map(x -> x.getId())
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/mappers/OrderMapper.java]: `x` is not informative

File: src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    public ShoppingCartMapper(UserService userService) {
+        this.userService = userService;
+    }
+
+    public ShoppingCartResponseDto getShoppingCartDto(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto shoppingCartResponseDto =
+                new ShoppingCartResponseDto();
+        shoppingCartResponseDto.setId(shoppingCart.getId());
+        shoppingCartResponseDto.setTicketIds(shoppingCart.getTickets().stream()
+                .map(x -> x.getId())
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/shop/service/UserService.java (hunk 9 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/service/UserService.java:11]: changed to getById everywhere

File: src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java (hunk 11 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java]: no

File: src/main/java/com/cinema/shop/model/dto/OrderResponseDto.java (hunk 13 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/model/dto/OrderResponseDto.java]: done

File: src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java (hunk 15 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/mappers/ShoppingCartMapper.java]: done

File: src/main/java/com/cinema/shop/service/TicketService.java (hunk 8 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/service/TicketService.java]: no, maybe, not now

File: src/main/java/com/cinema/shop/model/dto/UserResponseDto.java (hunk 7 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/model/dto/UserResponseDto.java]: ok

File: src/main/java/com/cinema/shop/model/dto/ShoppingCartResponseDto.java (hunk 6 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/model/dto/ShoppingCartResponseDto.java]: yes

File: src/main/java/com/cinema/shop/dao/TicketDao.java (hunk 4 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/dao/TicketDao.java]: nope

File: src/main/java/com/cinema/shop/controllers/OrderController.java (hunk 1 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/controllers/OrderController.java]: done

File: src/main/java/com/cinema/shop/controllers/UserController.java (hunk 3 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/controllers/UserController.java]: done

**Review by KseniiaMakarova — APPROVED** [review]
let's remove `getUserFromDto()` method from UserMapper - I believe you do not use it?

**Comment by margarita-ignatieva** [comment]
> let's remove `getUserFromDto()` method from UserMapper - I believe you do not use it?

no, don't

**Pull request merged**